src/dfx/src/main.rs

```diff
@@ -170,12 +174,35 @@ fn print_error_and_diagnosis(err: Error, error_diagnosis: Diagnosis) {
 }
 
 fn main() {
-    let cli_opts = CliOpts::parse();
+    let mut args = std::env::args_os().collect::<Vec<OsString>>();
+    let mut error_diagnosis: Diagnosis = NULL_DIAGNOSIS;
+
+    if let Ok(em) = ExtensionManager::new(dfx_version(), false) {
```

Comment:
Getting close!  This shouldn't swallow errors.  You could have `ExtensionManager::new()` return `Result<Option<..>>` or use some other means.